### ISSUE TITLE
Randomness in reference scripts with --test-only

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/vision#4587]
While running some experiments related to https://github.com/pytorch/vision/issues/4559, I tried the following:

```diff
diff --git a/references/classification/train.py b/references/classification/train.py
index a71d337a..1a429801 100644
--- a/references/classification/train.py
+++ b/references/classification/train.py
@@ -147,7 +147,7 @@ def load_data(traindir, valdir, args):
     print("Creating data loaders")
     if args.distributed:
         train_sampler = torch.utils.data.distributed.DistributedSampler(dataset)
-        test_sampler = torch.utils.data.distributed.DistributedSampler(dataset_test)
+        test_sampler = torch.utils.data.distributed.DistributedSampler(dataset_test, shuffle=False)
     else:
         train_sampler = torch.utils.data.RandomSampler(dataset)
         test_sampler = torch.utils.data.SequentialSampler(dataset_test)
```

```
torchrun --nproc_per_node=1 references/classification/train.py --model resnet18 --test-only --pretrained
```

I was hoping to get reproducible results across executions, i.e. always the same accuracy, but it seems like I still get a bit of variations across a few runs:

```
Test:  Acc@1 69.766 Acc@5 89.068
Test:  Acc@1 69.764 Acc@5 89.072
Test:  Acc@1 69.764 Acc@5 89.072
Test:  Acc@1 69.766 Acc@5 89.068
```

Does anyone know where this randomness might come from?

**Comment by NicolasHug** [comment]
BTW looks like the actual accuracy of resnet18 is a fair bit higher than what we report on the docs: 69.494 | 88.882
